### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by anderseknert**
* Mimic error message from OPA on hex decoding failure
* Use big integers in bit shift operations
* Enhanced type checking for minus built-in and regex.is_valid
* Set intersection of empty set should return empty set
* Use destructuring in parser functions

Also, made a few improvements to the compliance test suite, so that the generate form only is read by the test runner (and not linters, etc)..

### CODE REVIEW

File: src/main/clojure/jarl/parser.clj [hunk 1]
```diff
@@ -9,226 +9,170 @@
...
   (log/debug "making AssignVarOnceStmt stmt")
-  (let [source-index (get stmt-info "source")
-        target (get stmt-info "target")]
-    (fn [state]
-      (eval/eval-AssignVarOnceStmt source-index target state))))
+  (fn [state]
+    (eval/eval-AssignVarOnceStmt source target state)))
 
-(defn make-BlockStmt [stmt-info]
+(defn make-BlockStmt [{:strs [blocks] :as stmt-info}]
```
Comment (johanfylling) [inline src/main/clojure/jarl/parser.clj:32]: Destructuring in clojure is new to me. `stmt-info` is the original argument, and not an alias for `blocks`?

**Review by johanfylling — APPROVED** [review]
👍

File: project.clj [hunk 2]
```diff
@@ -11,13 +11,15 @@
   :main jarl.core
   :aot [jarl.core jarl.parser jarl.api]
   :direct-linking true
-  :aliases {"clj-kondo" ["with-profile" "+clj-kondo" "clj-kondo" "--lint" "src"]
+  :aliases {"clj-kondo" ["with-profile" "+clj-kondo" "clj-kondo" "--debug" "--lint" "src"]
```
[the remark this answers is not in the document]
Comment (anderseknert) [reply project.clj:14]: It prints the name of every file scanned, which I think was nice compared to nothing.

File: src/main/clojure/jarl/builtins/numbers.clj [hunk 3]
```diff
@@ -18,7 +19,9 @@
   ; (check-args (meta #'builtin-minus) a b)
   (if (and (set? a) (set? b))
     (set/difference a b)
-    (possibly-int (- a b))))
+    (if (and (number? a) (number? b))
+      (possibly-int (- a b))
+      (throw (errors/type-ex "minus: operand %s must be number but got %s" (if (number? b) 1 2) (types/java->rego (if (number? b) a b)))))))
```
[the remark this answers is not in the document]
Comment (anderseknert) [reply src/main/clojure/jarl/builtins/numbers.clj]: Thanks, fixed 👍

**Pull request merged**